src/Compiler/Microsoft.AspNetCore.Razor.Language/src/TagHelperBinder.cs

```diff
@@ -150,12 +148,7 @@ private void Register(TagHelperDescriptor descriptor)
                 _registrations[registrationKey] = descriptorSet;
             }
 
-            // We don't need to keep trying to add the same Descriptor to the HashSet for each matching rule.
-            if (!descriptorAdded)
-            {
-                _ = descriptorSet.Add(descriptor);
-                descriptorAdded = true;
-            }
+            _ = descriptorSet.Add(descriptor);
```

Comment:
Can we add a test that would catch the wrong behavior before this change?

Reply:
Yeah, I'm currently trying to concoct one. I wasn't sure if we needed to service this fix or not so wanted to get it out ASAP without tests to start with.